packages/manager/apps/web/client/app/hosting/database/order/public/components/steps/db-categories-offers/controller.js

```diff
@@ -17,7 +17,18 @@ export default class WebHostingDatabaseOrderComponentsDbCategoriesOffersControll
   }
 
   getDbEngines() {
-    return this.model.dbCategory?.selectVersion?.engines || [];
+    const engines = this.model.dbCategory?.selectVersion?.engines;
+    if (engines && engines.length) {
+      angular.forEach(engines, (engine) => {
+        engine.versions.sort((a, b) => {
+          // Order versions in descending order
+          return b.dbVersion.localeCompare(a.dbVersion, undefined, {
+            numeric: true,
+          });
+        });
+      });
+    }
+    return engines || [];
```

Comment:
Could be simplified. Also avoid using functions `angular.`, prefer native javascript functions instead

```suggestion
    const engines = this.model.dbCategory?.selectVersion?.engines || [];
    engines.forEach(({versions}) => {
        versions.sort((a, b) => {
          // Order versions in descending order
          return b.dbVersion.localeCompare(a.dbVersion, undefined, {
            numeric: true,
          });
        });
      });
    }
    return engines;
```

Reply:
```suggestion
      this.model.dbCategory?.selectVersion?.engines?.map((engine) => {
        engine.versions.sort((a, b) => {
          // Order versions in descending order
          return b.dbVersion.localeCompare(a.dbVersion, undefined, {
            numeric: true,
          });
        });
      }));
```

Reply:
Exact we can use map here because we change each engine versions sorting.